### PR TITLE
feat(discover-split): Update Discover, Alerts, Dashboards docs

### PR DESCRIPTION
**Pull request by narsaynorath**
In preparation for the dataset split in discover and dashboards, we need to update the docs to reflect this new change.

The changes typically involve removing `event.type:error` or `event.type:transaction` from being mentioned in search fields because after the dataset is selected, these are redundant. It also updates screenshots and calls out the dataset selector more explicitly.

### CODE REVIEW

**Review by shruthilayaj — COMMENTED** [review]
I wonder if we should split up this section
https://github.com/getsentry/sentry-docs/pull/11101/files#diff-0df0e661cacec6f97cfe0bc85f38b934a830010696797fbb2f93c927627af7e5R53 into explicit Errors and Transactions?
Can we also get @lizokm or someone from docs to review this?

**Comment by narsaynorath** [comment]
@shruthilayaj
> I wonder if we should split up this section

I was thinking about that as well, I tried to split them apart but I found myself duplicating the portion about being able to query for this data in Discover. I'll look at fleshing out the meaning of the datasets more and that should make it easier to split the sections.

**Comment by lizokm** [comment]
@narsaynorath thanks for updating. I will take a look at this tomorrow!

File: docs/product/explore/discover-queries/query-builder.mdx [hunk 1]
```diff
@@ -15,12 +15,17 @@ From the **Discover** page, you can build a query in three ways.
   2. Enter a new display name
   3. Click "Save"
 
-There are four main building blocks that impact the results of your saved query. You can use a combination of these to narrow down your search.
+There are five main building blocks that impact the results of your saved query. You can use a combination of these to narrow down your search.
```
Comment (lizokm) [inline docs/product/explore/discover-queries/query-builder.mdx:18]:
```suggestion
There are five main building blocks that impact the results of your saved query, a combination of which can be used to narrow down your search.
```

File: docs/concepts/search/searchable-properties/events.mdx [hunk 2]
```diff
@@ -8,7 +8,7 @@ Events are the underlying event data captured using Sentry SDKs — that is, err
 
 You can search by event properties in the following [sentry.io](https://sentry.io) pages:
 
-- Discover - in the query builder
+- Discover - in the query builder, depending on the dataset selection
 - Dashboards - within the widget builder, depending on dataset selection
```
Comment (lizokm) [inline docs/concepts/search/searchable-properties/events.mdx:12]:
```suggestion
- Discover - in the query builder (depending on the dataset selection)
- Dashboards - within the widget builder (depending on dataset selection)
```

File: docs/product/dashboards/widget-builder/index.mdx [hunk 3]
```diff
@@ -48,15 +48,19 @@ A big number visualization displays the current value of a single function. This
 
 ## Choose Your Dataset
 
-In the dataset selection step, choose which type of data you would like to use in your widget. This data is classified into two different datasets: [errors and transactions](#errors--transactions) and [issues](#issues).
+In the dataset selection step, choose which type of data you would like to use in your widget. This data is classified into four different datasets: [errors](#error), [transactions](#transactions), [issues](#issues), and [releases](#releases).
 
-### Errors & Transactions
+### Errors
+Choosing "Errors" allows you to query and aggregate error events in the same way you would for a [Discover Query](/product/explore/discover-queries/). This data is comprised of errors that occur in your application, which Sentry will use to group into issues. Some widget examples include:
```
Comment (lizokm) [inline docs/product/dashboards/widget-builder/index.mdx:54]:
```suggestion
Choosing "Errors" allows you to query and aggregate error events in the same way you would for a [Discover Query](/product/explore/discover-queries/). This data is comprised of errors that occur in your application, which Sentry will use to group into issues, for example:
```

File: docs/product/explore/discover-queries/query-builder.mdx [hunk 4]
```diff
@@ -15,12 +15,17 @@ From the **Discover** page, you can build a query in three ways.
   2. Enter a new display name
   3. Click "Save"
 
-There are four main building blocks that impact the results of your saved query. You can use a combination of these to narrow down your search.
+There are five main building blocks that impact the results of your saved query, a combination of which can be used to narrow down your search.
```
Comment (lizokm) [inline docs/product/explore/discover-queries/query-builder.mdx:18]:
```suggestion
There are five main building blocks that impact the results of your saved query, a combination of which can be used to narrow down your search:
```

File: docs/product/explore/discover-queries/query-builder.mdx [hunk 5]
```diff
@@ -15,12 +15,17 @@ From the **Discover** page, you can build a query in three ways.
...
 
-1. Project, environment, and date range filters
-2. Search Conditions
-3. Interactive Graph
-4. Table Columns
+1. Dataset selection
+2. Project, environment, and date range filters
+3. Search Conditions
+4. Interactive Graph
+5. Table Columns
```
Comment (lizokm) [inline docs/product/explore/discover-queries/query-builder.mdx:24]:
```suggestion
1. Dataset selection
2. Project, environment, and date range filters
3. Search conditions
4. Interactive graph
5. Table columns
```

**Review by lizokm — APPROVED** [review]
Thanks for making these updates!

**Comment by lizokm** [comment]
@narsaynorath should we merge this?

**Pull request merged**